test/backup.ts

```diff
@@ -0,0 +1,515 @@
+        backupId: BACKUP_NAME,
+        backup: {
+          name: BACKUP_FORMATTED_NAME,
+          database: DATABASE_FORMATTED_NAME,
+          expireTime: BACKUP_EXPIRE_TIME.toStruct(),
+        },
+      });
+
+      backup.request = config => {
+        assert.strictEqual(config.client, 'DatabaseAdminClient');
+        assert.strictEqual(config.method, 'createBackup');
+        assert.deepStrictEqual(config.reqOpts, expectedReqOpts);
+
+        assert.notStrictEqual(config.reqOpts, QUERY);
+        assert.deepStrictEqual(QUERY, ORIGINAL_QUERY);
+      };
+
+      await backup.create(DATABASE_FORMATTED_NAME, BACKUP_EXPIRE_TIME);
+    });
+
+    it('should accept gaxOptions and a callback', async () => {
+      const options = {
+        timeout: 1000,
+      };
+
+      backup.request = config => {
+        assert.deepStrictEqual(config.gaxOpts, options);
+      };
+
+      await backup.create(
+        DATABASE_FORMATTED_NAME,
+        BACKUP_EXPIRE_TIME,
+        options,
+        assert.ifError
+      );
+    });
+
+    describe('error', () => {
+      const REQUEST_RESPONSE_ARGS = [new Error('Error.'), null, null, null];
+
+      beforeEach(() => {
+        backup.request = (config, callback: Function) => {
+          callback.apply(null, REQUEST_RESPONSE_ARGS);
+        };
+      });
+
+      it('should execute callback with original arguments', done => {
+        backup.create(
+          DATABASE_FORMATTED_NAME,
+          BACKUP_EXPIRE_TIME,
+          (...args) => {
+            assert.deepStrictEqual(args, REQUEST_RESPONSE_ARGS);
+            done();
+          }
+        );
+      });
+    });
+  });
+
+  describe('getMetadata', () => {
+    const BACKUP_NAME = 'backup-name';
+
+    it('should make the correct request', async () => {
+      const QUERY = {};
+      const ORIGINAL_QUERY = extend({}, QUERY);
+      const expectedReqOpts = extend({}, QUERY, {
+        name: BACKUP_FORMATTED_NAME,
+      });
+
+      backup.request = config => {
+        assert.strictEqual(config.client, 'DatabaseAdminClient');
+        assert.strictEqual(config.method, 'getBackup');
+        assert.deepStrictEqual(config.reqOpts, expectedReqOpts);
+
+        assert.notStrictEqual(config.reqOpts, QUERY);
+        assert.deepStrictEqual(QUERY, ORIGINAL_QUERY);
+      };
+
+      await backup.getMetadata();
+    });
+
+    it('should accept gaxOpts and a callback', async () => {
+      const options = {
+        timeout: 1000,
+      };
+
+      backup.request = config => {
+        assert.deepStrictEqual(config.gaxOpts, options);
+      };
+
+      await backup.getMetadata(options, assert.ifError);
+    });
+
+    describe('error', () => {
+      const REQUEST_RESPONSE_ARGS = [new Error('Error.'), null];
+
+      beforeEach(() => {
+        backup.request = (config, callback: Function) => {
+          callback.apply(null, REQUEST_RESPONSE_ARGS);
+        };
+      });
+
+      it('should execute callback with original arguments', done => {
```

Comment:
Since we don't handle error and success paths differently, this test alone could be used. Just bump it up one level and delete the error/success describes 👍

Reply:
That makes sense. Although I chose to keep the test for the success case instead of the error case to make sure that the metadata info is returned correctly.